angularTest/src/app/course-list/course-list-item.interface.ts

```diff
@@ -1,9 +1,21 @@
 export interface CourseListItem {
-  creationDate: any | Date
+  creationDate: string
   id: string | number
   title: string
   date?: string
   description?: string
   top: boolean
   duration?: number
 }
+
+export class CCourseListItem implements CourseListItem {
```

Comment:
Could you please explain why you've created  class? Also I have question regarding it's naming. If you want to use classes than it's better to prefix interfaces with `I` and use class without any prefixes

Reply:
Also I would suggest something like this for you in future.
``` 
public fields: string[] = ['creationDate', 'id', ...];

constructor(config: CourseListItem) { 
this.fields.forEach((field) => this[field] = config[field]);
}
```

Something like this